agent/command/papertrail_trace.go

```diff
@@ -0,0 +1,213 @@
+)
+
+const (
+	productionPapertrailAddress = "https://papertrail.devprod-infra.prod.corp.mongodb.com"
+)
+
+var (
+	errMustSpecifyKeyID              = fmt.Errorf("must specify key_id")
+	errMustSpecifySecretKey          = fmt.Errorf("must specify secret_key")
+	errMustSpecifyProduct            = fmt.Errorf("must specify product")
+	errMustSpecifyVersion            = fmt.Errorf("must specify version")
+	errMustSpecifyAtLeastOneFilename = fmt.Errorf("must specify at least one filename")
+)
+
+type papertrailTrace struct {
+	Address   string   `mapstructure:"address" plugin:"expand"`
+	KeyID     string   `mapstructure:"key_id" plugin:"expand"`
+	SecretKey string   `mapstructure:"secret_key" plugin:"expand"`
+	Product   string   `mapstructure:"product" plugin:"expand"`
+	Version   string   `mapstructure:"version" plugin:"expand"`
+	Filenames []string `mapstructure:"filenames" plugin:"expand"`
+	WorkDir   string   `mapstructure:"work_dir" plugin:"expand"`
+	base
+}
+
+func (t *papertrailTrace) Execute(ctx context.Context,
+	comm agentinternalclient.Communicator, logger agentinternalclient.LoggerProducer, conf *agentinternal.TaskConfig) error {
+	if err := evergreenutil.ExpandValues(t, &conf.Expansions); err != nil {
+		return fmt.Errorf("apply expansions: %w", err)
+	}
+
+	client := &papertrailClient{
+		httpc:     &http.Client{},
+		address:   t.Address,
+		keyID:     t.KeyID,
+		secretKey: t.SecretKey,
+	}
+
+	task := conf.Task
+
+	papertrailBuildID := fmt.Sprintf("%s_%d", task.Id, task.Execution)
+
+	workdir := t.WorkDir
+	if workdir == "" {
+		workdir = conf.WorkDir
+	}
+
+	for _, file := range t.Filenames {
+		fullname := filepath.Join(workdir, file)
+
+		sha256sum, err := getSHA256(fullname)
+		if err != nil {
+			return fmt.Errorf("get sha256: %w", err)
+		}
+
+		// should already be the basename, but just in case
+		basename := filepath.Base(file)
+
+		args := traceArgs{
+			build:     papertrailBuildID,
+			platform:  "evergreen",
+			filename:  basename,
+			sha256:    sha256sum,
+			product:   t.Product,
+			version:   t.Version,
+			submitter: task.ActivatedBy,
+		}
+
+		if err := client.trace(ctx, args); err != nil {
+			return fmt.Errorf("run trace: %w", err)
+		}
+
+		logger.Task().Infof("Successfully traced '%s' (sha256://%s)", fullname, sha256sum)
+	}
+
+	return nil
+}
+
+func getSHA256(filename string) (string, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return "", fmt.Errorf("open file: %w", err)
+	}
+
+	defer f.Close()
+
+	h := sha256.New()
+	if _, err := io.Copy(h, f); err != nil {
+		return "", fmt.Errorf("read file: %w", err)
+	}
+
+	sha256sum := fmt.Sprintf("%x", h.Sum(nil))
+
+	return sha256sum, nil
+}
+
+func (t *papertrailTrace) ParseParams(params map[string]any) error {
+	conf := &mapstructure.DecoderConfig{
+		WeaklyTypedInput: true,
+		Result:           t,
+	}
+
+	decoder, err := mapstructure.NewDecoder(conf)
+	if err != nil {
+		return fmt.Errorf("new decoder: %w", err)
```

Comment:
I can do that, but pkg/errors is deprecated in favor of the stdlib errors package, any reason why you're still using the former?

Reply:
Uh maybe just because we didn't know that. Would still prefer to use that format for consistency and because of how it also returns a trace, which fmt doesn't do. Will open a ticket to consider if we should switch packages longer term.

Reply:
Updated everything to use github.com/pkg/errors instead!